src/sql/src/plan/hir.rs

```diff
@@ -2718,15 +2774,20 @@ impl VisitChildren<HirScalarExpr> for HirRelationExpr {
                     f(aggregate.expr.as_mut());
                 }
             }
-            Distinct { input: _ }
-            | TopK {
+            TopK {
                 input: _,
                 group_key: _,
                 order_key: _,
-                limit: _,
-                offset: _,
+                limit,
+                offset,
                 expected_group_size: _,
+            } => {
+                if let Some(limit) = limit {
+                    f(limit)
```

Comment:
It was a bug on `main` that `limit` was not visited here! (and the same is 2 more cases below)